include/dlaf/eigensolver/bt_reduction_to_band/impl.h

```diff
@@ -346,8 +346,10 @@ void BackTransformationReductionToBand<B, D, T>::call(
     }
 
     for (const auto& kj_panel : panelW2.iteratorLocal())
-      ex::start_detached(dlaf::comm::scheduleAllReduceInPlace(mpi_col_task_chain(), MPI_SUM,
-                                                              panelW2.readwrite_sender(kj_panel)));
+      ex::start_detached(
+          dlaf::comm::scheduleAllReduceInPlace(mpi_col_task_chain(), MPI_SUM,
```

Comment:
I would have expected this...what am I missing?

```suggestion
          dlaf::comm::scheduleAllReduceInPlace(ex::make_unique_any_sender(mpi_col_task_chain()), MPI_SUM,
```

Reply:
The first argument to the `scheduleX` functions is typically a `unique_any_sender<PromiseGuard<Communicator>>` which is concrete type (i.e. not templated): https://github.com/msimberg/DLA-Future/blob/8143864a536657496717d4ae0264527c8b5db7d3/include/dlaf/communication/kernels/all_reduce.h#L60. The `unique_any_sender` constructor is _not_ `explicit`: https://github.com/pika-org/pika/blob/ff04f4b9290af3fa58d384d52effb3469bd79d84/libs/pika/execution_base/include/pika/execution_base/any_sender.hpp#L751-L758, which means that the communicator can be constructed directly without using `make_unique_any_sender`.

What might be a bit confusing is that in some cases the communicator sender does have to be wrapped explicitly in `make_unique_any_sender`, like here: https://github.com/msimberg/DLA-Future/blob/8143864a536657496717d4ae0264527c8b5db7d3/include/dlaf/communication/broadcast_panel.h#L66. That's because in some cases the `scheduleX` function needs to support taking both a sender sending a plain `Communicator` and a `PromiseGuard<Communicator>` like here: https://github.com/msimberg/DLA-Future/blob/8143864a536657496717d4ae0264527c8b5db7d3/include/dlaf/communication/kernels/p2p.h#L38-L48. In those cases the communicator sender also becomes templated and does need the explicit `make_unique_any_sender`, just like the tile senders.

I've added only the minimally required `make_unique_any_sender` and left the cases where an implicit conversion can happen without `make_unique_any_sender` but I'm happy to be explicit as well if you prefer it. Something like this: https://github.com/eth-cscs/DLA-Future/pull/674#pullrequestreview-1211601021 would obviously reduce the need to be explicit because the conversion has already happened.

Full disclosure: I noticed that I left at least one set of kernels with a templated tile sender even though it doesn't need the templating: https://github.com/msimberg/DLA-Future/blob/8143864a536657496717d4ae0264527c8b5db7d3/include/dlaf/communication/kernels/broadcast.h#L75-L97. I think this was a leftover from earlier iterations when I needed it to be templated in some situations.

My questions to all of you:
1. do you prefer to explicitly use `make_unique_any_sender` everywhere even if it's not strictly needed?
2. do you prefer to not template the communicator sender if it's not currently needed?

Thanks for pointing this out @albestro!

Reply:
1. I prefer to use it only when is needed.
2. My idea is to "force" the use of pipelines for collectives, while point to point can use a plain sender. However I didn't think about all possible consequences yet. I suggest to keep the current version and discuss it in details later.